pyinfra/api/state.py

```diff
@@ -75,38 +80,46 @@ class State(object):
     Manages state for a pyinfra deploy.
     """
 
-    initialised = False
+    initialised: bool = False
 
     # A pyinfra.api.Inventory which stores all our pyinfra.api.Host's
-    inventory = None
+    inventory: Optional["Inventory"] = None
```

Comment:
As a lot of `state` variables are initialized with `None` at first, this will make Mypy complain a lot all around the codebase when passing the `state` and accessing these variables (such as `inventory` or `config` for example.

One option for the future is to add a property decorator to ensure this will exist when other functions will try to access some attribute, this should make accessing these attributes a bit safer.

Tried to keep this branch strictly related to typing though, so no further action on this.

Reply:
Just pushed up https://github.com/Fizzadar/pyinfra/pull/881/commits/947f8f691aec10e8ed963991db49afa36a58abc5 to address this - tests pass and the default `None`'s don't appear to be used anymore!

Was investigating moving things around so `Inventory.state` isn't required, but let's keep this branch strictly on typing and added https://github.com/Fizzadar/pyinfra/issues/884 to track that.